core/jazz_acl/components/validation.js

```diff
@@ -16,23 +16,45 @@
 
 const errorHandlerModule = require("./error-handler.js")();
 
-async function validateInput(userInput) {
-
-		if (!userInput || !userInput.method) {
-			throw (errorHandlerModule.throwInputValidationError("Invalid or missing arguments"));
+function validateBasicInput(userInput) {
+    if(!userInput) {
+      throw (errorHandlerModule.throwInternalServerError("Input parameters are missing"));
+    }
+		if (!userInput.method) {
+			throw (errorHandlerModule.throwInputValidationError("Method is missing"));
     }
 
-    if (!userInput.resourcePath) {
+    if (!userInput.path) {
 			throw (errorHandlerModule.throwInputValidationError("Missing the resource path"));
 		}
 
 		if (!userInput.principalId) {
 			throw (errorHandlerModule.throwForbiddenError("You aren't authorized to access this resource"));
 		}
+}
+
+function validateGetPoliciesInput(userInput) {
+  if (!userInput.query.serviceId) {
+    throw (errorHandlerModule.throwInputValidationError("Service Id is missing"));
+  }
+}
+
+function validatePostPoliciesInput(userInput) {
+  if (!userInput.body) {
+    throw (errorHandlerModule.throwInternalServerError("Body is missing"));
+  }
+
+  if (!userInput.body.serviceId) {
+    throw (errorHandlerModule.throwInputValidationError("Service Id is missing"));
+  }
 
-		return true;
+  if (!userInput.body.policies) {
```

Comment:
Should we extend this and validate each policy object?  I see that we are using policy.userId, policy.category & policy.permission directly else where. It's better if we check for these k/v pairs if policies[] is not empty.

Reply:
This is difficult to validate as policies can be empty for the same POST (add or delete)